internal/handlers/irc/irc.go

```diff
@@ -14,14 +13,15 @@ and the IRCSettings that were passed into NewClient
 type Client struct {
 	*girc.Client
 	Settings internal.IRCSettings
+	verbose internal.Debug
 	sendToTg func(string)
 }
 
 /*
 NewClient returns a new IRCClient based on the provided settings
 */
-func NewClient(settings internal.IRCSettings) Client {
-	fmt.Println("Creating new IRC bot client...")
+func NewClient(settings internal.IRCSettings, debug internal.Debug) Client {
```

Comment:
You'll also need to change the `NewClient` function signatures to accept the interface instead